packages/astro/src/core/config/schema.ts

```diff
@@ -636,19 +653,15 @@ export function createRelativeSchema(cmd: string, fileProtocolRoot: string) {
 		),
 	})
 		.transform((config) => {
-			// If the user changed outDir but not build.server, build.config, adjust so those
-			// are relative to the outDir, as is the expected default.
-			if (
-				!config.build.server.toString().startsWith(config.outDir.toString()) &&
-				config.build.server.toString().endsWith('dist/server/')
-			) {
-				config.build.server = new URL('./dist/server/', config.outDir);
-			}
+			// If the user changed `outDir`, we need to also update `build.client` and `build.server`
```

Comment:
Do we need to do the individual transforms above? Maybe we could always transform in this transform instead. I feel like we duplicate the work